### PR TITLE
fix(<select />): fix select undefined state

### PR DESCRIPTION
**Pull request by robertmihaiman**

### CODE REVIEW

File: src/components/Select/Select.tsx [hunk 1]
```diff
@@ -10,25 +10,28 @@ import './Select.scss';
 
 export type SelectableOption = { value: string; label: string; disabled?: boolean };
 
-export type SelectProps = Omit<React.HTMLAttributes<HTMLDivElement>, 'value' | 'onChange'> & {
+export type SelectProps = Omit<
+  React.HTMLAttributes<HTMLDivElement>,
+  'defaultValue' | 'value' | 'onChange'
```
Comment (hevele-moda) [inline src/components/Select/Select.tsx:15]: Is it required to omit these? AFAIK these are not on `React.HTMLAttributes<HTMLDivElement>`.

File: src/components/Select/Select.tsx [hunk 2]
```diff
@@ -135,8 +138,10 @@ export const Select: React.FC<SelectProps> = ({
       className={classNames('Select', { 'Select--disabled': disabled }, className)}
       ref={selectRef}
       {...rest}
+      onChange={undefined}
```
Comment (hevele-moda) [inline src/components/Select/Select.tsx:141]: Is this required here?

File: src/components/Select/Select.tsx (hunk 1 above)
Comment (robertmihaiman) [reply src/components/Select/Select.tsx:15]: well, if not, onChange would look like

(property) onChange?: (((event: React.FormEvent<HTMLDivElement>) => void) & ((value: string) => void)) | undefined

Comment (robertmihaiman) [reply src/components/Select/Select.tsx:15]: but now it's like

`(JSX attribute) onChange?: ((value: string) => void) | undefined`

Comment (hevele-moda) [reply src/components/Select/Select.tsx:15]: I see, but should `value` and `defaultValue` be omitted too?

Comment (robertmihaiman) [reply src/components/Select/Select.tsx:15]: yes, we would like to limit those from

`defaultValue?: string | number | string[];`

to `string | undefined`

Comment (hevele-moda) [reply src/components/Select/Select.tsx:15]: ok, it's weird that `React.HTMLAttributes<HTMLDivElement>` has those

**Pull request merged**